src/networking.c

```diff
@@ -3627,6 +3641,11 @@ static int parseMultibulk(client *c,
                 *argv = zrealloc(*argv, sizeof(robj *) * (*argv_len));
             }
 
+            /* Check that what follows argv is a real \r\n */
+            if (c->querybuf[c->qb_pos + c->bulklen] != '\r' || c->querybuf[c->qb_pos + c->bulklen + 1] != '\n') {
```

Comment:
I saw that this when I refactored this code in #2092 but I was worried about affecting performance if we add more checks. Maybe we should add `unlikely` annotation to help the branch prediction?

```suggestion
            if (unlikely(c->querybuf[c->qb_pos + c->bulklen] != '\r' ||
                         c->querybuf[c->qb_pos + c->bulklen + 1] != '\n')) {
```

Reply:
Why do we need this check? (Maybe It is late I missed it when I reviewed) Like we already verified the first CRLF so when will we hit this case?

Reply:
It is after reading the bulk content. We already know the length here so we just  read the content of fixed length

    $LENGTH\r\nCONTENT\r\n
                      ^
                      |
                    We are here

Reply:
Oh Yeh. I get it now. I think the same branching hint can be applied to the other checks as well then, although IMO it will have negligible impact.